### PR TITLE
Throw error when ledger db is incompatible

### PR DESCRIPTION
**Pull request by Jasagredo**
# Description

Previously, this would fail with an difficult to decipher error. Now it is much more understandable

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/BackingStore.hs [hunk 1]
```diff
@@ -182,6 +182,14 @@ newTVarBackingStore lookup_ rangeRead_ forwardValues_ enc dec initialization = d
...
       (slot, values) <- case initialization of
         Left (FS.SomeHasFS fs, BackingStorePath path) -> do
+          dir <- FS.unsafeToFilePath fs path
+          tvarFileExists <- FS.doesFileExist fs (extendPath path)
+          unless
+            tvarFileExists
+            (error $ "In-Memory database not found in the database directory: "
+                   <> show dir
+                   <> ".\nPre-UTxO-HD and LMDB implementations are incompatible with the In-Memory \
+                      \ implementation. Please delete your ledger database directory." )
```
Comment (jorisdral) [inline ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/BackingStore.hs:192]: Possibly use `Exn.throw` with a custom datatype?

**Review by jorisdral — APPROVED** [review]
LGTM, but I'm wondering whether we should have a backingstore-independent approach that checks if a snapshot is of the desired backingstore-type before trying to initialise a store. In the current approach, each backing store implementation would have the responsibility of doing the check

File: ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/BackingStore.hs (hunk 1 above)
Comment (Jasagredo) [reply ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/BackingStore.hs:192]: Sounds reasonable

Comment (Jasagredo) [reply ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/BackingStore.hs]: Done!

**Comment by Jasagredo** [comment]
> backingstore-independent approach that checks if a snapshot is of the desired backingstore-type before trying to initialise a store

Probably this would be the right way to go, indeed.

**Pull request merged**